datalad/distribution/utils.py

```diff
@@ -214,7 +215,43 @@ def _install_subds_from_flexible_source(ds, sm_path, sm_url, reckless):
     # do fancy update
     if sm_path in ds.get_subdatasets(absolute=False, recursive=False):
         lgr.debug("Update cloned subdataset {0} in parent".format(subds))
+        # TODO: move all of that into update_submodule ??
+        # TODO: direct mode ramifications?
```

Comment:
I'm too late obviously, but for FTR: Yes, it should. `AnnexRepo ` makes sure, that all the git calls are done with `-c core.bare=False` automatically in direct mode. It needs to override it only if this isn't sufficient, but in most cases it is.